### PR TITLE
fix(cli): resolve real path of entry file for `export:embed`

### LINKED ISSUE #28890: [eas-build-local] Metro errors with SHA-1 for file ... index.js ... is not computed when using "index.js" as package main
### Minimal reproducible example

https://github.com/fabiovieira1901/app.git

### What platform(s) does this occur on?

iOS

### Did you reproduce this issue in a development build?

Yes

### Summary

With Expo version 49, I was able to run the command "eas build --platform ios --local" without any issues. After updating to version 51, this command now consistently results in an error:

error: SHA-1 for file /var/folders/gc/f5p8tj751dvgf363hqcj5d500000gn/T/eas-build-local-nodejs/cf95be6e-e9e7-49d5-ba6e-53c60e47c9d5/build/apps/app-milene-martins/index.js (/var/folders/gc/f5p8tj751dvgf363hqcj5d500000gn/T/eas-build-local-nodejs/cf95be6e-e9e7-49d5-ba6e-53c60e47c9d5/build/apps/app-milene-martins/index.js) is not computed.

### Environment

expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.5
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 18.18.1 - ~/.nvm/versions/node/v18.18.1/bin/node
      Yarn: 1.22.19 - /opt/homebrew/bin/yarn
      npm: 9.8.1 - ~/.nvm/versions/node/v18.18.1/bin/npm
    Managers:
      CocoaPods: 1.14.3 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.5, iOS 17.5, macOS 14.5, tvOS 17.5, visionOS 1.2, watchOS 10.5
    IDEs:
      Android Studio: 2021.2 AI-212.5712.43.2112.8609683
      Xcode: 15.4/15F31d - /usr/bin/xcodebuild
    npmPackages:
      expo: ~51.0.6 => 51.0.6 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.74.1 => 0.74.1 
    npmGlobalPackages:
      eas-cli: 9.0.5
      expo-cli: 6.3.10
    Expo Workflow: managed

### Expo Doctor Diagnostics

✔ Check Expo config for common issues
✔ Check package.json for common issues
✔ Check native tooling versions
✔ Check dependencies for packages that should not be installed directly
✔ Check for common project setup issues
✔ Check for issues with metro config
✔ Check npm/ yarn versions
✔ Check Expo config (app.json/ app.config.js) schema
✔ Check that packages match versions required by installed Expo SDK
✔ Check that native modules do not use incompatible support packages
✔ Check for legacy global CLI installed locally
✔ Check that native modules use compatible support package versions for installed Expo SDK

Didn't find any issues with the project!

### PR DESCRIPTION
**Pull request by byCedric**
# Why

Fixes #28890

# How

Since we probably want to backport this, I implemented the fix very defensively (only fixing this exact use case).

TL;DR; When running `eas build --local`, the project is copied to a temporary system folder (e.g. `/var/folders/...`, or `/tmp/...`). When running `expo export:embed`, the CWD is set to `/private/var/folders/...` since this is the real location, but the entry file is referred to as `/var/folders`, messing up Metro's haste map in the process.

[See this comment for more info](https://github.com/expo/expo/issues/28890#issuecomment-2115291182)

# Test Plan

### Before this fix
- `$ git clone git@github.com:brentvatne/repro-28890.git ./test-28890`
- `$ cd ./test-28890 && yarn`
- `$ eas build --local` should fail with "SHA-1 not calculated"

### After this fix
- `$ git clone git@github.com:brentvatne/repro-28890.git ./test-28890`
- `$ cd ./test-28890 && yarn`
- `$ eas build --local` should succeed

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/@expo/cli/src/export/embed/exportEmbedAsync.ts [hunk 1]
```diff
@@ -228,3 +228,17 @@ export async function exportEmbedAssetsAsync(
...
 }
+
+/**
+ * This is a workaround for Metro not resolving entry file paths to their real location.
+ * When running exports through `eas build --local` on macOS, the `/var/folders` path is used instead of `/private/var/folders`.
+ *
+ * See: https://github.com/expo/expo/issues/28890
+ */
+function resolveRealEntryFilePath(projectRoot: string, entryFile: string): string {
+  if (projectRoot.startsWith('/private') && entryFile.startsWith('/var')) {
```
Comment (byCedric) [inline packages/@expo/cli/src/export/embed/exportEmbedAsync.ts:239]:
```suggestion
  if (projectRoot.startsWith('/private/var') && entryFile.startsWith('/var')) {
```

**Review by brentvatne — APPROVED** [review]
